synapse_data_warehouse/synapse/dynamic_tables/V2.23.0__accessrequirement_latest.sql

```diff
@@ -0,0 +1,26 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE DYNAMIC TABLE IF NOT EXISTS ACCESSREQUIREMENT_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+    WITH latest_unique_rows AS (
+        SELECT
+            accessrequirementsnapshots.*,
+        FROM
+            {{database_name}}.synapse_raw.accessrequirementsnapshots --noqa: TMP
+        WHERE
+            CHANGE_TYPE != 'DELETE'
+        AND
+            SNAPSHOT_TIMESTAMP >= CURRENT_TIMESTAMP - INTERVAL '14 DAYS'
+        QUALIFY ROW_NUMBER() OVER (
+                PARTITION BY id
+                ORDER BY change_timestamp DESC, snapshot_timestamp DESC
```

Comment:
Why do we order primarily by `change_timestamp`? Isn't `snapshot_timestamp` sufficient? Or could there be a case where a create/update/delete event occurred but there was no snapshot?

Reply:
Thanks, that makes sense.